pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -36,47 +38,68 @@ import (
 type topologyPairToScore map[string]map[string]int64
 
 type podAffinityPriorityMap struct {
-	// nodes contain all nodes that should be considered.
-	nodes []*v1.Node
-	// tracks a topology pair score so far.
-	topologyScore topologyPairToScore
+	topologyScore         topologyPairToScore
+	pod                   *v1.Pod
+	affinityTerms         []*processedWeightedAffinityTerm
+	antiAffinityTerms     []*processedWeightedAffinityTerm
+	sharedLister          schedulerlisters.SharedLister
+	hardPodAffinityWeight int32
 	sync.Mutex
 }
 
-func newPodAffinityPriorityMap(nodes []*v1.Node) *podAffinityPriorityMap {
-	return &podAffinityPriorityMap{
-		nodes:         nodes,
-		topologyScore: make(topologyPairToScore),
-	}
+type processedWeightedAffinityTerm struct {
+	namespaces  sets.String
+	selector    labels.Selector
+	weight      int32
+	topologyKey string
 }
 
-func (p *podAffinityPriorityMap) processTerm(term *v1.PodAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, weight int64) error {
-	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(podDefiningAffinityTerm, term)
+func getProcessedWeightedAffinityTerm(pod *v1.Pod, term *v1.PodAffinityTerm, weight int32) (*processedWeightedAffinityTerm, error) {
```

Comment:
can't accept WeightedPodAffinityTerm as a parameter because of https://github.com/kubernetes/kubernetes/pull/85959/files#diff-ac123860716bbbb4ed0f26cc27062bb7R201